testing/typescript/data/custom_store.ts

```diff
@@ -0,0 +1,39 @@
+/* eslint-disable @typescript-eslint/no-unsafe-member-access */
+/* eslint-disable @typescript-eslint/explicit-function-return-type */
+// eslint-disable-next-line import/no-extraneous-dependencies
+import $ from 'jquery';
+import { HttpClient } from '@angular/common/http';
+import CustomStore from '../../../js/data/custom_store';
+
+export function loadByFetch(url: string): CustomStore {
+  return new CustomStore({
+    load: () => fetch(url).then((response) => response.json()),
+  });
+}
+
+export function loadByFetchWithModification(url: string): CustomStore {
+  return new CustomStore({
+    load: () => fetch(url)
+      .then((response) => response.json())
+      .then((data) => ({
+        data: data.data,
+        totalCount: data.totalCount,
+        summary: data.summary,
+        groupCount: data.groupCount,
+      })),
+  });
+}
+
+export function loadByAjax(url: string, data: string): CustomStore {
+  return new CustomStore({
+    load: (): JQueryXHR => $.ajax({
+      url, cache: false, dataType: 'json', data,
+    }),
+  });
+}
+
+export function loadByHttpClientAngular(http: HttpClient, url: string): CustomStore {
+  return new CustomStore({
+    load: () => http.get(url).toPromise(),
+  });
+}
```

Comment:
We can and **should** write integrational tests so that we could track if third-parties developers introduce BC's to our API. But:

1.  The syntax should be always the same (no return type, no args, informative name)
2.  We should add a unit test in addition to integrational test to avoid BC's in our code. In the case below:
    - the integrational test adds a context for behavior
    - the unit test fixes the behavior)

```suggestion
export function loadAcceptsAngularHttpClient() {
  const http: HttpClient;

  new CustomStore({
    load: () => http.get('url').toPromise(),
  });
}

export function loadAcceptsPromiseOfObject() {
  new CustomStore({
    load: () => new Promise(r => r({} as object))
  });
}
```

Reply:
Or
```
load: () => new Promise<object>(undefined)
```